spark-shell/src/main/scala/org/apache/mahout/sparkbindings/shell/MahoutSparkILoop.scala

```diff
@@ -48,55 +77,63 @@ class MahoutSparkILoop extends SparkILoop {
 
     conf.set("spark.executor.memory", "1g")
 
-    sparkContext = mahoutSparkContext(
+    _interp.sparkContext = mahoutSparkContext(
       masterUrl = master,
       appName = "Mahout Spark Shell",
       customJars = jars,
       sparkConf = conf
     )
 
-    echo("Created spark context..")
+    echoToShell("Created spark context..")
     sparkContext
   }
 
+  // need to change our SparkDistributedContext name to 'sc' since  we cannot override the
+  // private sparkCleanUp() method.
+  // this is technically not part of Sparks explicitly defined Developer API though
+  // nothing in the SparkILoopInit.scala file is marked as such.
   override def initializeSpark() {
-    intp.beQuietDuring {
-      command("""
+    _interp.beQuietDuring {
+      _interp.interpret("""
 
-         @transient implicit val sdc: org.apache.mahout.math.drm.DistributedContext =
+         @transient implicit val sc: org.apache.mahout.math.drm.DistributedContext =
```

Comment:
the problem is that the clean up method that we were using is not part of the Spark REPL Developer API.  So the method that we were using to close the SparkContext:
```
def sparkCleanUp() {
      echoToShell("Stopping Spark context.")
      _interp.beQuietDuring {
        _interp.interpret("sdc.stop()")
      }
   }
```

is not available to us to override anymore.  So upon exiting the shell, the private `SparkIloop.sparkCleanUp()` is called and looks to call `sc.close()`: 
  
```
private def sparkCleanUp(){
    echo("Stopping spark context.")
    intp.beQuietDuring {
      command("sc.stop()")
    }
  }
 ```

and giving an error,  leaving our `sdc` context open.